api/clients/config.go

```diff
@@ -0,0 +1,49 @@
+package clients
+
+import (
+	"fmt"
+	"time"
+)
+
+type EigenDAClientConfig struct {
+	// RPC is the HTTP provider URL for the Data Availability node.
+	RPC string
+
+	// The total amount of time that the client will spend waiting for EigenDA to confirm a blob
+	StatusQueryTimeout time.Duration
+
+	// The amount of time to wait between status queries of a newly dispersed blob
+	StatusQueryRetryInterval time.Duration
+
+	// The total amount of time that the client will waiting for a response from the EigenDA disperser
+	ResponseTimeout time.Duration
+
+	// The quorum IDs to write blobs to using this client. Should not include quorums 0 or 1.
+	CustomQuorumIDs []uint
+
+	// Signer private key in hex encoded format. This key should not be associated with an Ethereum address holding any funds.
+	SignerPrivateKeyHex string
+
+	// Whether to disable TLS for an insecure connection when connecting to a local EigenDA disperser instance.
+	DisableTLS bool
+
+	// The blob encoding version to use when writing blobs from the high level interface.
+	PutBlobEncodingVersion BlobEncodingVersion
+}
+
+var DefaultQuorums = map[uint]bool{0: true, 1: true}
+
+func (c *EigenDAClientConfig) Check() error {
```

Comment:
These validations and default parameter setup are usually done in constructors.
Maybe we can turn this into a constructor?
`func NewEigenDAClientConfig(...) (*EigenDAClientConfig, error)`

Reply:
Having the `Check()` method shares the same disadvantages as you'd need to pass in all fields to construct the config + calling `Check` method is completely optional and easy to miss by the user. 
By making the fields private, callers are forced to run the validation/set the default values.

Reply:
The client calls Check() in case the user forgets. The client can't call NewEigenDAClientConfig() if the user forgets.